packages/2d/src/curves/BezierSegment.ts

```diff
@@ -0,0 +1,69 @@
+import {Segment} from './Segment';
+import {CurveSampler} from './CurveSampler';
+import {Vector2} from '@motion-canvas/core/lib/types';
+import {UniformCurveSampler} from './UniformCurveSampler';
+import {BezierCurve} from './BezierCurve';
+import {moveTo} from '../utils';
+
+export abstract class BezierSegment extends Segment {
+  protected readonly length: number;
+  protected readonly pointSampler: CurveSampler;
+
+  protected constructor(protected readonly bezier: BezierCurve) {
+    super();
+    this.length = this.bezier.arcLength;
+    this.pointSampler = new UniformCurveSampler(this.bezier);
+  }
+
+  public get arcLength(): number {
+    return this.length;
+  }
+
+  public get points(): Vector2[] {
+    return this.bezier.points;
+  }
+
+  public getPoint(distance: number): [Vector2, Vector2] {
+    const closestPoint = this.pointSampler.pointAtDistance(distance);
+    return [closestPoint.position, closestPoint.tangent];
+  }
+
+  public transformPoints(matrix: DOMMatrix): Vector2[] {
+    return this.points.map(point => point.transformAsPoint(matrix));
+  }
+
+  public draw(
+    context: CanvasRenderingContext2D | Path2D,
+    start = 0,
+    end = 1,
+    move = true,
+  ): [Vector2, Vector2, Vector2, Vector2] {
+    const startDistance = this.length * start;
+    const endDistance = this.length * end;
+    const startPoint = this.pointSampler.pointAtDistance(startDistance);
+    const endPoint = this.pointSampler.pointAtDistance(endDistance);
+    const endPointT = (endPoint.t - startPoint.t) / (1 - startPoint.t);
+
+    const [, startSegment] = this.bezier.split(startPoint.t);
+    const [segment] = startSegment.split(endPointT);
+    const points = segment.points;
```

Comment:
I think you could skip a lot of work by checking if `start === 0 && end === 1`
Seems like that would be the case most of the time, and no splitting is necessary in that case.

Reply:
Good point. Changed.